### DISCUSSION TITLE
how to get a component of the applied shear stress

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#27114] [Q&A Modules: Solid mechanics] [answered]
I need the first component of the applied shear stress and its type is `MaterialProperty<std::vector<Real>>` as I see in CrystalPlasticityKalidindiUpdate. Then, I need to finally compute the element's average value in the postprocessor. I learned that the AuxKernel MaterialRealVectorValueAux is what I need. But I got the error: 
```
The following error occurred in the object "shear_stress0", of type "MaterialRealVectorValueAux".

The requested non-AD material property 'uncracked_applied_shear_stress' of type 'libMesh::VectorValue<double>'
is already retrieved or declared as a non-AD property of type 'std::vector<double>' 
```
I know this is because of different type as `std::vector<Real>` instead of `libMesh::VectorValue<double>`. But how can I fix this? Or I must write my own auxkernel?

**Comment by GiudGiud** [comment]
Hello

~~just prefix the auxkernel withAD to use the AD version~~

**Reply by RWTHLHK** [reply]
thanks for your reply. But I still got the error: 
```
*** ERROR ***
The following error occurred in the object "shear_stress0", of type "ADMaterialRealVectorValueAux".

The requested AD material property 'uncracked_applied_shear_stress' of type` 'libMesh::VectorValue<double>'`
is already retrieved or declared as a non-AD property of type `'std::vector<double>`'
```
I think the reason is `MaterialRealVectorValueAux` and `ADMaterialRealVectorValueAux` only support `'libMesh::VectorValue<double>'`. And I checked the source code and this is indeed the case.

**Reply by RWTHLHK** [reply]
I tried many methods and all failed. This is not a difficult task but it is so hard to do in MOOSE......

**Reply by cpgr** [reply, marked as answer]
Sounds like the material is a vector in the c++ container sense, not a vector in the mathematical sense. You should use MaterialStdVectorAux instead of MaterialRealVectorAux

**Reply by RWTHLHK** [reply]
thanks!
